### DISCUSSION TITLE
Does the editor work with Typescript files?

### DISCUSSION THREAD

**Discussion by Seanmclem** [suren-atoyan/monaco-react#145] [Q&A] [answered]
I had trouble in the past using react Monaco editors with typescript files. Basically the editor wouldn't understand the syntax without additional hacky changes that were way beyond me. I was hoping this Library would work with typescript files, but I see no mention in the read me so. How is the compatibility with typescript files?

**Comment by Seanmclem** [comment, marked as answer]
It does seem to have some level of Typescript support out of the box. Specifically when you use `language="typescript"` on the <Editor /> tag.

**Reply by suren-atoyan** [reply]
Hello @Seanmclem, yes, you are correct. It has Typescript support out of the box, and yes, you should use `language="typescript"` for that. You can check more about supported languages in the [original repo](https://github.com/Microsoft/monaco-editor). Plus you can play with different examples in different languages in the [playground](https://monaco-react.surenatoyan.com/)
